txpipe/twopoint.py

```diff
@@ -353,15 +410,27 @@ def get_lens_catalog(self, data, i):
             ra = data['ra'][mask]
             dec = data['dec'][mask]
 
-        cat = treecorr.Catalog(
-            ra=ra, dec=dec,
-            ra_units='degree', dec_units='degree')
+        if self.config['var_methods']=='jackknife':
+            cat = treecorr.Catalog(
+                ra=ra, dec=dec,
+                ra_units='degree', dec_units='degree',
+                npatch=self.config['npatch'])
+        else:
+            cat = treecorr.Catalog(
+                ra=ra, dec=dec,
+                ra_units='degree', dec_units='degree')
 
         if 'random_bin' in data:
             random_mask = data['random_bin']==i
-            rancat  = treecorr.Catalog(
-                ra=data['random_ra'][random_mask], dec=data['random_dec'][random_mask],
-                ra_units='degree', dec_units='degree')
+            if self.config['var_methods']=='jackknife':
+                rancat  = treecorr.Catalog(
+                    ra=data['random_ra'][random_mask], dec=data['random_dec'][random_mask],
+                    ra_units='degree', dec_units='degree',
+                    npatch=self.config['npatch'])
```

Comment:
This won't work.  And is possibly why Joe is (sometimes?) seeing problems with the patches matching up, as mentioned on Slack.  This command makes a whole new set of patches for the random that don't correspond to the patches used for the lenses.

You want instead
```
rancat  = treecorr.Catalog(
                    ra=data['random_ra'][random_mask], dec=data['random_dec'][random_mask],
                    ra_units='degree', dec_units='degree',
                    patch_centers=cat.patch_centers)
```
This will identify the right objects in the random that correspond to each patch used by `cat`.

Reply:
Do we need the same patches for the lens and source catalogs also, and across the different tomographic bins?

Reply:
We probably should have, if you note the initial pull request, we note that there might be a potential issue with consistency of the patches between catalogs and that needs improvement.

Reply:
Only if you want to make the combined covariance matrix across tomo bins.  (Which I suspect is the case!)

Reply:
Yes, we do indeed!  @empEvil @jpratmarti - you might need to have one process generate the patches and then mpi bcast them to the others.

Reply:
I'd recommend restructuring to have the full catalog make the patches, and then use those patch_centers for all the tomo bins.

Reply:
My suggestion would be to add in a pipeline stage that does what Mike suggests, and I think we were planning to do that. I thought we were waiting on a newer version of treecorr though

Reply:
I don't think you want an entirely separate stage - then it would have to read the data in all over again  Since each process has the entire data set right now anyway, having one process make the patches on the full catalog would be fine.  Not sure if that will change in future. 

Alternatively, we do have a counts map by this point made in the mapping stage, so you could use that to construct the patch centres and share them.  That could be its own stage, since loading the map is is much faster than loading the catalog.

Reply:
You might want to look at how I did it in this script:
https://github.com/rmjarvis/TreeCorr/pull/99/files#diff-0b1e108387239363fa6a206cf6ba9e5aR61
I load in just a portion of the full catalog for making the patches, and then write those patches to a file.  Then the other catalogs can use those patch_center files when they read in.  This may be more efficient, since it can load less data into memory when building up each patch.